modules/aws-kernel/src-js/smithy4s/aws/CryptoPlatformCompat.scala

```diff
@@ -50,9 +44,9 @@ trait CryptoPlatformCompat {
 }
 
 object CryptoPlatformCompat {
-  val sha256 = "sha256"
-  val utf8 = "UTF-8"
-  val hex = "hex"
-  val binary = "binary"
-  val ascii = "ascii"
+  final val sha256 = "sha256"
```

Comment:
weird, I'd have thought that was implied in static objects. What's the effect of adding final here ? 

Also, do you happen to have a `list of do's/don'ts for Scala library maintainers` somewhere ?

Reply:
I learned this one from Seb :) per spec the Scala compiler treats these specially.

> A constant value definition is of the form
> ```scala
> final val x = e
> ```
> where `e` is a [constant expression](https://www.scala-lang.org/files/archive/spec/2.13/06-expressions.html#constant-expressions). The final modifier must be present and no type annotation may be given. References to the constant value `x` are themselves treated as constant expressions; in the generated code they are replaced by the definition's right-hand side `e`.

https://www.scala-lang.org/files/archive/spec/2.13/04-basic-declarations-and-definitions.html#value-declarations-and-definitions

Reply:
I think this has an impact on the inferred type too, as `final` fields are singleton types. BuildInfo also has an option to turn this on btw

Reply:
not sure but maybe a type annotation would work _if_ it's the singleton type 🤔